### PR TITLE
add slim-scroll w/ git auto-update

### LINKED ISSUE #9971: [Request] Add slimscroll.js
**Library name:** slimscroll.js
**Git repository url:** https://github.com/kamlekar/slim-scroll
**npm package url(optional):** 
**License(s):**: MIT
**Official homepage:**:  http://uxrim.com/slim-scroll
**Wanna say something? Leave message here:**

=====================
Notes from cdnjs maintainer:
Please read the README.md and CONTRIBUTING.md document first.

You are welcome to add a library via sending pull request,
it'll be faster than just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by sufuf3**
close #9971, cc @kamlekar
@kennynaoh please help me review this PR, thank you.

Pull request for issue: #9971 
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/kamlekar/slim-scroll
 * Official website (optional, not the repository): http://uxrim.com/slim-scroll
 * NPM package url (optional):
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 4
   - Count of stars: 101
   - Count of forks: 19
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by maruilian11** [comment]
since the author has added the minified file, please remove the map file to match the git repo file structure.

**Comment by sufuf3** [comment]
@maruilian11 thanks for your review.
I remove the map file, please help me review this PR again, thank you.

**Comment by kamlekar** [comment]
Hi. Thanks for considering my request. FYI, the min version I provided isn't updated one and it is hard for me to maintain a min version for every release.

**Comment by sufuf3** [comment]
@kamlekar so should we still minify the js file when we update the newest version?

We use uglify-js to generate the min.js and map files.
https://github.com/PeterDaveHello/web-minify-helper/blob/f287dacdbc7b5674c7cb4e49cc6673bcedb404b4/minify.sh

**Comment by kamlekar** [comment]
Yes please, if that can be automated. Thanks.

**Comment by sufuf3** [comment]
@kamlekar thank you for your infomation.
@maruilian11 I modified the file again, please help me review this PR, thank you.

**Comment by PeterDaveHello** [comment]
@kamlekar For the integrity, I prefer the minification to be done at official side.

**Comment by kamlekar** [comment]
@PeterDaveHello Currently, I haven't considered any particular compression technique. The available minification was done through some online platform. I need to look into the available compression techniques, then chose one and automate the same in my repo. Which might take long time and depends on my available time. So, I thought to do the minification from your side, if it is hassle-free. Thanks.

**Comment by PeterDaveHello** [comment]
@kamlekar Not a problem, if you don't mind, we can also try to help implement on your side.

**Comment by kamlekar** [comment]
@PeterDaveHello That's amazing :D. Thanks. Right now I am in office but if it takes less time, I am ready to implement it now with your support.

**Comment by PeterDaveHello** [comment]
@kamlekar I'll take a look later, we can accept this PR by now.

**Comment by PeterDaveHello** [comment]
@sufuf3 I think you can send a PR to help add `package.json` to upstream first.

File: ajax/libs/slim-scroll/package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    "url": "https://github.com/kamlekar/slim-scroll.git"
+  },
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/kamlekar/slim-scroll.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "slimscroll.js"
```
Comment (maruilian11) [inline ajax/libs/slim-scroll/package.json]: use `slimscroll*` here.

**Comment by sufuf3** [comment]
@maruilian11  Imodified it, please help me review this PR, thank you.

**Comment by sufuf3** [comment]
@pvnr0082t I added the old versions, please help me review this PR again, thank you.

**Comment by maruilian11** [comment]
Please only add versions **before** v1.2. 
Don't add versions already can be got from auto-update.

**Comment by sufuf3** [comment]
@maruilian11 sorry about that.
I modified it please help me review this PR again, thank you.

**Comment by maruilian11** [comment]
thanks, please also provide the reference of the old versions(where did you get them?).

**Comment by sufuf3** [comment]
@maruilian11  @pvnr0082t  thanks for your review, I changed the commit description. 
Please help me review this PR again, thank you.

**Comment by maruilian11** [comment]
where is the reference?

**Comment by kennynaoh** [comment]
@sufuf3 Should we add css file? https://github.com/kamlekar/slim-scroll/blob/v1.2/resources/slimscroll.css#L1

**Comment by kamlekar** [comment]
@kennynaoh CSS file is not needed

**Comment by kennynaoh** [comment]
@kamlekar Thanks a lot.
@sufuf3 Please remove the css file.

**Comment by sufuf3** [comment]
@kennynaoh I remove the css file.
Please help me review this PR again, thank you.

**Comment by kennynaoh** [comment]
Hello @kamlekar I found that there is a official minified file for `slimscroll.js` in the master branch but not in latest version tag. Could you also add the official minified file for `slimscroll.js` in the latest version tag or maybe the next release? Then we can automatically add it by using auto-update config. Thanks!

**Comment by kamlekar** [comment]
@kennynaoh I think it is better to ignore that available minified version and provide one from your end. I had stated this [before](https://github.com/cdnjs/cdnjs/pull/10469#issuecomment-276274577).

Thanks.

**Comment by kennynaoh** [comment]
@kamlekar Thanks!

**Comment by sufuf3** [comment]
@pvnr0082t I modified it.
Please help me review this PR again, thank you.

**Comment by PeterDaveHello** [comment]
@sufuf3 where is you pull request to upstream for package.json? I thought I mention that for more than 2 weeks: https://github.com/cdnjs/cdnjs/pull/10469#issuecomment-276310398

**Comment by kamlekar** [comment]
Hi @sufuf3, I released new version of [slim-scroll](https://github.com/kamlekar/slim-scroll) but unable to see the same update in CDNjs. Hope it will get updated after some time. I am worried because I reshuffled the directory structure. Thanks.

**Comment by PeterDaveHello** [comment]
@kamlekar it's because you changed the directory structure, ref: https://github.com/cdnjs/cdnjs/blob/master/ajax/libs/slim-scroll/package.json#L17-L28

**Comment by PeterDaveHello** [comment]
Fixed.

**Comment by kamlekar** [comment]
Ah ok. Sorry about that. The structure needed change to keep things organised.

**Comment by PeterDaveHello** [comment]
@kamlekar yeah no worries, feel free to directly send a PR for it next time when you need it.

**Pull request merged**